### PR TITLE
discocss: make discord target binary configurable

### PR DESCRIPTION
**Pull request by Kranzes**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/discocss/default.nix [hunk 1]
```diff
@@ -17,6 +18,15 @@ stdenvNoCC.mkDerivation rec {
     sha256 = "sha256-afmQCOOZ1MwQkbZZNHYfq2+IRv2eOtYBrGVHzDEbUHw=";
   };
 
+  patches = lib.optional discordAlias [
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/discocss/default.nix:21]: If discordAlias is false then this is a noop, right? then we shouldn't apply the patch conditionally.

Comment (Kranzes) [reply pkgs/tools/misc/discocss/default.nix:21]: Sure, but why?

Comment (Kranzes) [reply pkgs/tools/misc/discocss/default.nix:21]: Either don't hurt.

Comment (SuperSandro2000) [reply pkgs/tools/misc/discocss/default.nix:21]: To make sure the patch applies with the next update.

Comment (Kranzes) [reply pkgs/tools/misc/discocss/default.nix:21]: It wont. And the patch needs to be removed in the next update.

Comment (Kranzes) [reply pkgs/tools/misc/discocss/default.nix:21]: Considering the PR will be merged for the next update that is.

**Comment by JavaCafe01** [comment]
@Kranzes Your [PR](https://github.com/mlvzk/discocss/pull/17) has been merged, and there has been a version bump to [0.2.1](https://github.com/mlvzk/discocss/releases/tag/v0.2.1), so I think all we need now is a version bump for the package in nixpkgs.

**Pull request closed without merging**